cpp/dolfinx/fem/Expression.h

```diff
@@ -143,17 +143,33 @@ class Expression
 
   /// @brief Evaluate Expression on cells.
   /// @param[in] mesh Cells on which to evaluate the Expression.
-  /// @param[in] cells Cells on which to evaluate the Expression.
+  /// @param[in] entities List of entities to evaluate the expression on. This
```

Comment:
I think it would be nicer if we construct using entity indices, not (cell index, facet index). Most of our public user API is based on entity indices, e.g. locate_entities_foo, pack_coefficients, MeshTags.

Reply:
This is not trivial, and out of scope of this PR. The rationale for this is that you might want to compute an expression on an internal facet and you would then need to determine what side you should evaluate one. 

I agree that it would be good to expose general packing functionality to the user (also for custom integration measures, like one-sided interior facet integrals).